hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/deployment/ClientJobSubmissionTest.java

```diff
@@ -0,0 +1,187 @@
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.deployment;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.config.JetConfig;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.core.JobStatus;
+import com.hazelcast.jet.impl.JetClientInstanceImpl;
+import com.hazelcast.jet.impl.JobSummary;
+import com.hazelcast.jet.pipeline.JournalInitialPosition;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.Sources;
+import com.hazelcast.nio.Address;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class ClientJobSubmissionTest extends JetTestSupport {
+
+    @Rule
+    public final Timeout timeoutRule = Timeout.seconds(360);
+
+    private JetInstance client;
+    private JetInstance instance;
+
+    @Before
+    public void setUp() {
+        JetConfig jetConfig = new JetConfig();
+        jetConfig.getHazelcastConfig().getMapEventJournalConfig("journal*").setEnabled(true);
+        instance = createJetMember(jetConfig);
+        JetInstance jetInstance = createJetMember(jetConfig);
+        Address address = jetInstance.getCluster().getLocalMember().getAddress();
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().setSmartRouting(false);
+        clientConfig.getGroupConfig().setName(jetConfig.getHazelcastConfig().getGroupConfig().getName());
+        clientConfig.getNetworkConfig().getAddresses().clear();
+        clientConfig.getNetworkConfig().getAddresses().add(address.getHost() + ":" + address.getPort());
+        client = createJetClient(clientConfig);
+    }
+
+    @Test
+    public void Given_nonSmartClient_When_jobSubmitted_Then_executedSuccessfully() {
+        //Given
+        String sourceName = "source";
+        String sinkName = "sink";
+        fillListWithInts(instance.getList(sourceName), 10);
+
+        //When
+        Pipeline p = Pipeline.create();
+        p.drawFrom(Sources.list(sourceName))
+         .drainTo(Sinks.list(sinkName));
+        client.newJob(p).join();
+
+        //Then
+        assertEquals(10, instance.getList(sinkName).size());
+    }
+
+    @Test
+    public void Given_nonSmartClient_When_jobSubmitted_Then_jobCanBeFetchedByIdOrName() {
+        //Given
+        String jobName = randomName();
+
+        //When
+        Pipeline p = Pipeline.create();
```

Comment:
I would refactor the pipeline to a common method since it's used in a lot of the tests